### PR TITLE
Cover more termination cases

### PR DESCRIPTION
**Pull request by Andarist**
I've started from wanting to provide a test for #1 and when doing so I've noticed several problems with the actual implementation and while fixing it I've also encountered some "edge cases" that I had to fix too.

1. `sSrc` was subscribed **each time** `src` has emitted **anything**
2. `sink` was notified (`sink(t, d)`) from `sSrc` callbag, while it should be notified from `src` callbag
3. upward termination (done by `sink`) was just passed to the source, while it should also be handled internally by `takeUntil` and passed upwards afterwards

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,22 +1,40 @@
 const takeUntil = sSrc => src => (start, sink) => {
-  let talkback, sTalkback;
+  if (start !== 0) return;
+  let sourceTalkback, sTalkback;
+  let inited = false;
```
Comment (Andarist) [inline index.js:4]: I'm hating this `initied`/`done` dance but we have to postpone `sink(2)` until it's actually greeted to cover sync pullable sources used as notifier

**Comment by Andarist** [comment]
@franciscotln friendly 🏓

**Comment by Andarist** [comment]
Thanks ❤️ !
